tests/plugins/test_fetch_sources.py

```diff
@@ -300,3 +301,32 @@ def test_id_and_nvr_mismatch(self, requests_mock, docker_tasker, koji_session, t
             runner.run()
         msg = 'When specifying both an id and an nvr, they should point to the same image build'
         assert msg in str(exc.value)
+
+    @pytest.mark.parametrize(('build_type', 'koji_build_nvr'), [
+        ('rpm', 'foobar-1-1'),
+        ('module', 'foobar-1-1'),
+        ('image', 'foobar-source-1-1'),
+    ])
+    def test_invalid_source_build(self, requests_mock, docker_tasker, koji_session, tmpdir,
+                                  build_type, koji_build_nvr):
+        mock_koji_manifest_download(requests_mock)
+        runner = mock_env(tmpdir, docker_tasker, koji_build_nvr=koji_build_nvr, koji_build_id=1)
+
+        name, version, release = koji_build_nvr.rsplit('-', 2)
+        koji_build = {'build_id': 1, 'nvr': koji_build_nvr, 'name': name,
+                      'version': version, 'release': release}
+
+        flexmock(koji_session).should_receive('getBuild').and_return(koji_build)
+        flexmock(koji_session).should_receive('getBuildType').and_return({build_type: {}})
```

Comment:
it would be nice to add a test case for operator images:
```
{'image': {}, 'operator-manifests': {'}}
```

Reply:
sure